components/layout_2020/flow/mod.rs

```diff
@@ -1324,25 +1366,27 @@ fn solve_inline_margins_for_in_flow_block_level(
         },
         (LengthOrAuto::LengthPercentage(start), LengthOrAuto::Auto) => (start, free_space - start),
         (LengthOrAuto::LengthPercentage(start), LengthOrAuto::LengthPercentage(end)) => {
+            let free_space = Length::zero().max(free_space - start - end);
+            justification = justify_self_alignment(containing_block, free_space);
```

Comment:
Does setting a margin on one side of a block take precedence over `<center>`? It might be worth leaving a comment to that effect here.

Reply:
It doesn't take precedence, both work. When not avoiding floats, this centers the border box within the margin-shrunken containing block (equivalent to centering the margin box within the containing block). This matches other browsers and the spec. But of course, `auto` margins take all the free space, so no centering is possible.

When avoiding floats, this centers the border box within the intersection of the margin-shrunken containing block and the float-shrunken containing block. This matches Firefox. The spec is wrong (https://github.com/w3c/csswg-drafts/issues/9963), WebKit is broken, and Blink is nonsensical.

Reply:
I have added some comments.